app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -447,6 +449,23 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         }
     }
 
+    @VisibleForTesting
+    internal fun subscribeForNewAddonsIfNeeded(
+        scope: CoroutineScope = CoroutineScope( Dispatchers.IO )
+    ) {
+        scope.launch {
+            val haveUnsupportedAddons =
+                components.addonManager.getAddons().any { it.isInstalled() && !it.isSupported() }
+            if (haveUnsupportedAddons) {
+                components.supportedAddonsChecker.registerForChecks()
+            } else {
+                // As checks are a persistent subscriptions, we have to me sure
+                // we remove any previous subscriptions.
+                components.supportedAddonsChecker.unregisterForChecks()
+            }
```

Comment:
@csadilek before I adding tests I just want to confirm with you that changing the checks from being **sync** to be **async** is not a problem. The change needs to made to be able to query  add-ons from the add-on manager.

Reply:
The switch to sync/async would not be a problem, but the unconditional call to `getAddons` is. Our motivation for doing this is to reduce the number of network requests. With this logic as is we would still call getAddons on every startup, potentially hitting the network (cache is only 48 hours). So, we need to come up with something better here. Either remember that we already made the decision and not check again for that user or better: Let's do this in `onExtensionsLoaded` above and check against the extension metadata for "disabled as unsupported". This way we won't ever hit the AMO endpoint for this check.